dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx

```diff
@@ -62,55 +70,68 @@ export default function DepositBTCModal() {
     [onError, handlePause],
   )
 
-  const { sendBitcoinTransaction, transactionHash } = useDepositBTCTransaction(
-    onDepositBTCSuccess,
-    onDepositBTCError,
-  )
+  const { mutate: sendBitcoinTransaction, status } = useMutation({
+    mutationKey: ["send-bitcoin"],
+    mutationFn: async ({
+      recipient,
+      amount,
+    }: {
+      recipient: string
+      amount: bigint
+    }) => {
+      if (!address) {
+        throw new Error("Bitcoin account was not connected.")
+      }
 
-  useEffect(() => {
-    if (transactionHash) {
-      dispatch(setTxHash(transactionHash))
-    }
-  }, [dispatch, transactionHash])
+      if (!connector) {
+        throw new Error("Connector was not defined.")
+      }
+      // @ts-expect-error adjust types to handle bitcoin wallet wrappers
+      const client = await connector.getClient()
+
+      const satoshis = Number(amount)
+
+      // @ts-expect-error adjust types to handle bitcoin wallet wrappers
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-assignment
+      const txhash: string = await client.sendBitcoin(recipient, satoshis)
+      return txhash
+    },
+    onSuccess: onDepositBTCSuccess,
+    onError: onDepositBTCError,
+  })
```

Comment:
We could keep this code in `useDepositBTCTransaction`.

Reply:
https://github.com/thesis/acre/pull/821/commits/1440e0796c9e8561e2fb8b22ac3a0c482670cd18